roles/confluent.common/tasks/secrets_protection.yml

```diff
@@ -51,25 +61,32 @@
     secrets_protection_masterkey: "{{ slurped_masterkey.content|b64decode}}"
   when: not secrets_protection_masterkey
 
-- name: Encrypt All Properties that Contain 'password'
-  shell: |
-    {{ confluent_cli_path }} secret file encrypt --config-file {{ config_path }} \
-      --local-secrets-file {{ secrets_file }} \
-      --remote-secrets-file {{ secrets_file }}
-  environment:
-    CONFLUENT_SECURITY_MASTER_KEY: "{{ secrets_protection_masterkey }}"
-  when: encrypt_passwords|bool
+- name: Create Encyrpt Properties List
+  set_fact:
+    encryrpt_properties: "{{ final_properties | dict2items | map(attribute='key') | select('match', properties_patterns|join('|'))
+      + properties if encrypt_passwords|bool else properties }}"
+  vars:
+    properties_patterns:
```

Comment:
Reading the PR description, I get the impression that the user should be allowed to modify this list. Aren't you missing a variable for that in `confluent.variables` role?

Reply:
ya so you get two variables
```
kafka_broker_secrets_protection_encrypt_passwords: true/false # defaults to true
kafka_broker_secrets_protection_encrypt_properties: [list,of,properties] # defaults to []
```

So if you say encrypt_passwords, then the list gets everything in the pattern list + the user provided property list (which may be empty)

If you don't say encrypt_passwords then this list will only have your user provided property list.

Thinking about it now, I should apply the unique filter after combining the two lists

Reply:
But what if you don't want to encrypt some properties matching the "default patterns"? Wouldn't it be better to have a default, and allow for a complete override?

Reply:
if you dont want to encrypt the properties matching the "default patters" just set 
```
kafka_broker_secrets_protection_encrypt_passwords: false
```
and set
```
kafka_broker_secrets_protection_encrypt_properties: [the, props, you, do, want]
```

Reply:
Hmmm, isn't that a bit confusing? 😀 I would expect "encrypt properties" to control this. "Encrypt password" is a bit unprecise to me. The term password could mean a lot of things. IMO the term secret is better. https://martinfowler.com/bliki/TwoHardThings.html

Reply:
ya, so for more context. its super easy to turn secrets protection on. and with the initial secrets protection release, if you turned it on you'd get all props containing password masked. At the time, I knew that was insufficient, and gave the additional option to add on top of that.

Now i'm trying to make things better by masking more by default (hence the pattern list)

I also don't want to introduce new variables... so i just updated the docs. I do agree the var names aren't the best haha

I feel the best user experience is you just turn it on and dont have to include any additional props bc cp-ansible is smart enough to mask the correct props. This pr is heading in that direction.